### PR TITLE
Izhuer

### PR DESCRIPTION
**Pull request by ZhangZhuoSJTU**
# Commit for the RSoC

## What I have done

+ Rewrite CSE
+ Build Raw Memory SSA generation

__However__, because I have changed the algorithm for Memory SSA generation this week, it build failed for the lifetime, I will fix this tomorrow at lateset.

Sorry for delay and gap in the first month, because of many competitions and my laziness, I will pick the progress this month。

## What's the next step

+ Value Set Analysis
+ Type Interface Analysis

Now, the memory ssa is only raw memory ssa, which is lack of the type information, my next step will move on it, and maybe finish it at the end of Aug.

### CODE REVIEW

**Comment by chinmaydd** [comment]
Hey @ZhangZhuoSJTU , please make changes mentioned in #65 before confirming failure of tests due to your changes.

**Comment by ZhangZhuoSJTU** [comment]
@chinmaydd  Ok, I will do it tomorrow, for it's 12 o'clock now, hah.

File: minidec/main.rs [hunk 1]
```diff
@@ -101,8 +101,7 @@ fn main() {
             let mut cse = CSE::new(&mut rfn.ssa);
             cse.run();
         }
-        {
-            // Value Set Analysis
+        if false {
```
Comment (XVilka) [inline minidec/main.rs]: So you disabled VSA for now, right?

Comment (ZhangZhuoSJTU) [inline minidec/main.rs]: yep, this VSA only concentrates on x86/x64 and cause some panics, which I don't why. So I disabled it temporarily.

**Comment by ZhangZhuoSJTU** [comment]
@chinmaydd  Actually, I cannot finger out the reason why builds fail. seems weird. Do you know why?

**Comment by chinmaydd** [comment]
I faced a similar issue with my original PR as well. I can't seem to figure out why those specific tests are overclocking. Specifically, `bin_ls_x86_idioms` and `bin_ls_grep_replace` are timing out.  I'll try to look into it as well.

**Comment by ZhangZhuoSJTU** [comment]
grep_and_replace is the key point, I am going deep in src/analysis/matcher/gmatch.rs.

**Pull request merged**